Misc/NEWS.d/next/C API/2024-04-08-09-44-29.gh-issue-117534.54ZE_n.rst

```diff
@@ -0,0 +1,2 @@
+Add checking for input parameter in :func:`iso_to_umd` to fix the assertion
+in :func:`days_before_year` due to wrong ISO date string.

```

Comment:
Generally speaking we try to document the public-facing behavior of a patch, not the implementation details. Here's how I would word this:

```suggestion
Improve validation logic in the C implementation of :meth:`datetime.fromisoformat` to better handle invalid years. 
```

It is also customary to add "Patch by <Your name>" at the end of the news fragment, but that is not required, and I don't know your name, so I cannot include it in the suggestion.

Reply:
Thank you for the comments. The text is updated.